server/src/main/java/org/opensearch/cluster/routing/WeightedRoutingService.java

```diff
@@ -154,4 +158,15 @@ public void verifyAwarenessAttribute(String attributeName) {
             throw validationException;
         }
     }
+
+    public void ensureNoOngoingDecommissionAction(ClusterState state) {
+        DecommissionAttributeMetadata decommissionAttributeMetadata = state.metadata().decommissionAttributeMetadata();
+        if (decommissionAttributeMetadata != null && decommissionAttributeMetadata.status().equals(DecommissionStatus.FAILED) == false) {
```

Comment:
What if status is `SUCCESSFUL`?

Reply:
Only when Decommission has failed, we can allow the weight update can go through. Not for any other status

Reply:
Isn't `SUCCESSFUL` a final state? The error message of this `if` condition says: `a decommission action is ongoing with status...`

Reply:
A succesful decommission state means the transport execution has completed. The nodes are decommissioned in this state (not able to start pre voting or join). Hence DecommissionAction ongoing means a zone is decommissioned and the cluster rejects pings from decommissioned node